src/lib.rs

```diff
@@ -225,6 +227,17 @@ impl DeviceTree {
         })
     }
 
+    /// Load a device tree from a raw pointer. This should be used when the size of the device tree
+    /// is unknown
+    pub fn load_from_raw_pointer(addr: *const u8) -> Result<DeviceTree, DeviceTreeError> {
```

Comment:
Please mark this function as `unsafe`. Otherwise it is okay to merge.

Reply:
I'm not used to ``unsafe fn`` why should this function be considered unsafe? Because we are careful by checking the magic number in the header, is it "that" unsafe?

I will do it, I'm just wondering why

Reply:
It uses a raw pointer, and from OS perspective, reading an unknown pointer might lead to unknown behavior.

Reply:
Ok